test/error_messages.coffee

```diff
@@ -1317,3 +1317,24 @@ test "can't use pattern matches for loop indices", ->
     a for b, {c} in d
              ^^^
   '''
+
+test "bare 'super' is no longer allowed", ->
+  assertErrorFormat 'class extends A then constructor: -> super', '''
+    [stdin]:1:35: error: unexpected ->
+    class extends A then constructor: -> super
+                                      ^^
+  '''
+
+test "soaked 'super' in constructor", ->
+  assertErrorFormat 'class extends A then constructor: -> super?()', '''
+    [stdin]:1:38: error: Unsupported reference to 'super'
```

Comment:
What about _illegal_ instead of _unsupported?_ (The latter almost sounds like this is a TODO comment, doesn't it?)